### PR TITLE
Transaction data displaying using metadata

### PR DESCRIPTION
**Pull request by relatko**
Transaction metadata are now stored in the app as `uint8_t[]`. These metadata are now used to display all signTx validation screens (parser.c).
Several minor improvements to SCREEN macro in getPubkey call validating screens (addr.c) to match improvements done made for signTx.

### CODE REVIEW

File: app/src/parser.c [hunk 1]
```diff
@@ -411,44 +407,145 @@ parser_error_t parser_printAuthorizer(const flow_proposal_authorizer_t *v,
...
+//Their use should be hiden behind SCREEN macro
+#define GET_NUM_ITEMS_DISPLAY_IDX_STARTING_VALUE (-1)
+
+parser_error_t parser_getItem_internal(int8_t *displayIdx,
+                                       char *outKey, uint16_t outKeyLen,
+                                       char *outVal, uint16_t outValLen,
+                                       uint8_t pageIdx, uint8_t *pageCount) {
+    zemu_log_stack("parser_getItem_internal");
+
+    //validate contract
```
Comment (janmazak) [inline app/src/parser.c:427]: This is the same as in `addr.c`?

File: app/src/tx_metadata.c [hunk 2]
```diff
@@ -103,23 +995,23 @@ parser_error_t parseTxMetadata(uint8_t scriptHash[SCRIPT_HASH_SIZE], const NV_VO
     #define READ_CHAR(where) \
         { \
             if (!(parsed < txMetadataLength)) { \
-                return PARSER_TEMPLATE_ERROR; \
+                return PARSER_METADATA_ERROR; \
             } \
             (*where) = txMetadata[parsed++]; \
```
Comment (janmazak) [inline app/src/tx_metadata.c:1000]: Shouldn't `where` be in parentheses?

File: app/src/tx_metadata.h [hunk 3]
```diff
@@ -32,12 +32,13 @@ typedef enum {
...
 } argument_type_e;
 
-#define MAX_TEMPLATE_NUMBER_OF_HASHES 10
-#define MAX_TEMPLATE_STRING_LENGTH 100
+#define PARSER_MAX_ARGCOUNT 10
 
 //It is planed that all these strings may be on flash, thus they are volatile (for NanoX and NanoSPlus)
 typedef struct {
     argument_type_e argumentType;
+    uint8_t arrayMinElements; //defined only for ARGUMENT_TYPE_ARRAY and ARGUMENT_TYPE_OPTIOALARRAY
```
Comment (janmazak) [inline app/src/tx_metadata.h:40]:
```suggestion
    uint8_t arrayMinElements; //defined only for ARGUMENT_TYPE_ARRAY and ARGUMENT_TYPE_OPTIONALARRAY
```

**Review by Wolog2021 — APPROVED** [review]

**Comment by pgebheim** [comment]
@Wolog2021 Same question here, when will this be merged?

Should I organize these all within one of our epics?

**Pull request merged**